sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/implementation/SpringServiceImpl.java

```diff
@@ -283,7 +358,54 @@ public SpringServiceImpl withoutCertificate(String name) {
         return this;
     }
 
+    @Override
+    public SpringServiceImpl withGitConfig(String uri, String branch, List<String> filePatterns) {
+        return withGitConfigRepository(Constants.DEFAULT_TANZU_COMPONENT_NAME, uri, branch, filePatterns);
+    }
+
+    @Override
+    public SpringServiceImpl withGitConfigRepository(String name, String uri, String branch, List<String> filePatterns) {
+        if (CoreUtils.isNullOrEmpty(name)) {
+            return this;
+        }
+        this.gitRepositoryMap.computeIfAbsent(name, key ->
+            new ConfigurationServiceGitRepository()
+                .withName(name)
+                .withUri(uri)
+                .withPatterns(filePatterns)
+                .withLabel(branch)
+        );
+        updateConfigurationServiceTask = true;
+        return this;
+    }
+
+    @Override
+    public SpringServiceImpl withGitConfig(ConfigurationServiceGitProperty gitConfig) {
+        gitRepositoryMap.clear();
+        if (gitConfig != null && CoreUtils.isNullOrEmpty(gitConfig.repositories())) {
+            for (ConfigurationServiceGitRepository repository : gitConfig.repositories()) {
+                this.gitRepositoryMap.put(repository.name(), repository);
+            }
+        }
+        updateConfigurationServiceTask = true;
+        return this;
+    }
+
+    private void prepareCreateOrUpdateConfigurationService() {
+        List<ConfigurationServiceGitRepository> repositories = new ArrayList<>(this.gitRepositoryMap.values());
+        this.configurationServices.prepareCreateOrUpdate(new ConfigurationServiceGitProperty().withRepositories(repositories));
+    }
+
     private boolean isInUpdateMode() {
         return !isInCreateMode();
     }
+
+    boolean isEnterpriseTier() {
+        return innerModel().sku() != null && SkuName.E0.toString().equals(innerModel().sku().name());
+    }
+
+    private void clearCache() {
+        this.gitRepositoryMap.clear();
+        this.configurationServices.clear();
```

Comment:
Do we reset `updateConfigurationServiceTask` as well?

Reply:
Yes, I'll add it.